src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer.Common/BackgroundDocumentGenerator.cs

```diff
@@ -275,64 +275,84 @@ private void ProjectSnapshotManager_Changed(object? sender, ProjectChangeEventAr
         {
             case ProjectChangeKind.ProjectAdded:
                 {
-                    var projectSnapshot = args.Newer!;
-                    foreach (var documentFilePath in projectSnapshot.DocumentFilePaths)
+                    var newProject = args.Newer.AssumeNotNull();
```

Comment:
I kinda like this, but will probably take time to get used to. I would have just done `Assumes.NotNull(newProject)` on the next line

Reply:
Yeah, I was getting sick of adding extra lines. It's functionally equivalent to `Assume.NotNull()`.